jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/util/NameHelper.java

```diff
@@ -286,13 +289,22 @@ private String createFullFieldName(String nodeName, String prefix, String suffix
         return returnString;
     }
 
-    private String makeUnique(String className, JPackage _package) {
+    private String makeUnique(String className, JClassContainer container) {
         try {
-            JDefinedClass _class = _package._class(className);
-            _package.remove(_class);
+            JDefinedClass _class = container._class(className);
+            if (container instanceof JPackage) {
+                ((JPackage) container).remove(_class);
+            } else {
+                Iterator<JDefinedClass> iter = container.classes();
+                while (iter.hasNext()) {
+                    if (iter.next().equals(_class)) {
+                        iter.remove();
```

Comment:
Not completely happy about it but, unfortunately, JClass does not expose a method to remove a class.